AsyncFixer.Test/UnnecessaryAsyncTests.cs

```diff
@@ -327,6 +331,82 @@ Task Test2Async(string str)
             VerifyCSharpFix(test, fixtest);
         }
 
+        [Fact]
+        public void AwaitExpressionsUnderLambda()
+        {
+            var test = @"
+using System;
+using System.Threading.Tasks;
+
+class Program
+{
+    async Task foo()
+    {
+        await Task.Run(async () => await Task.FromResult(true));
+    }
+}";
+
+            var expected = new DiagnosticResult { Id = DiagnosticIds.UnnecessaryAsync };
+            VerifyCSharpDiagnostic(test, expected);
+
+            var fixtest = @"

```

Comment:
If the code shouldn't change then can't it use the same variable for the fix test too?

Reply:
We can, but we do not need to call `VerifyCSharpFix` with the same variable for no-warning cases. When there are no diagnostics reported, Roslyn will not call the fixer.